app/services/search_service.rb

```diff
@@ -113,6 +113,38 @@ def tagNearbyNodes(query, tag, limit = 10)
     end
   end
 
+  # Search nearby people with respect to given latitude, longitute and tags
+  # and package up as a DocResult
+  def tagNearbyPeople(query, tag, limit = 10)
+    raise("Must separate coordinates with ,") unless query.include? ","
+
+    lat, lon =  query.split(',')
+
+    user_locations = User.where('rusers.status <> 0')\
+                         .joins(:user_tags)\
+                         .where('value LIKE ?', 'lat:' + lat[0..lat.length - 2] + '%')\
+                         .distinct
+    if tag.present?
+      user_locations = User.joins(:user_tags)\
+                       .where('user_tags.value LIKE ?', tag)\
+                       .where(id: user_locations.select("rusers.id"))
+    end
+
+    ids = user_locations.collect(&:id).uniq || []
+
+    items = User.where('rusers.status <> 0').joins(:user_tags)
+                .where('rusers.id IN (?) AND value LIKE ?', ids, 'lon:' + lon[0..lon.length - 2] + '%')
```

Comment:
Hey 😄 !
How about using limit method over here ?
```suggestion
                .where('rusers.id IN (?) AND value LIKE ?', ids, 'lon:' + lon[0..lon.length - 2] + '%').limit(limit)
```

Reply:
I want to use limit after removing users that have location:blurred tag. That makes sense? But I see I've added limit method in the wrong place! Can you take a look now?

To be honest I wasn't sure if I really need to filter users with location:blurred tag, I'm just doing because other methods are doing so. Can an user have lat/lon tags and location:blurred at the same time?

Reply:
Yes @milaaraujo , we can have lat/lon tags and location:blurred tags at the same time .
This signifies that a user wants to give a rough idea of his location rather than giving exact location , which is why we need to filter users with location:blurred tag .
Thank you !